src/um/wincrypt.rs

```diff
@@ -1334,279 +1745,146 @@ pub const szOID_PROTECTED_PROCESS_SIGNER: &'static str = "1.3.6.1.4.1.311.10.3.2
-pub const CERT_CHAIN_POLICY_ALLOW_UNKNOWN_CA_FLAG: DWORD = 0x00000010;
-pub const CERT_CHAIN_POLICY_IGNORE_WRONG_USAGE_FLAG: DWORD = 0x00000020;
-pub const CERT_CHAIN_POLICY_IGNORE_INVALID_NAME_FLAG: DWORD = 0x00000040;
-pub const CERT_CHAIN_POLICY_IGNORE_INVALID_POLICY_FLAG: DWORD = 0x00000080;
-pub const CERT_CHAIN_POLICY_IGNORE_END_REV_UNKNOWN_FLAG: DWORD = 0x00000100;
-pub const CERT_CHAIN_POLICY_IGNORE_CTL_SIGNER_REV_UNKNOWN_FLAG: DWORD = 0x00000200;
-pub const CERT_CHAIN_POLICY_IGNORE_CA_REV_UNKNOWN_FLAG: DWORD = 0x00000400;
-pub const CERT_CHAIN_POLICY_IGNORE_ROOT_REV_UNKNOWN_FLAG: DWORD = 0x00000800;
-pub const CERT_CHAIN_POLICY_IGNORE_ALL_REV_UNKNOWN_FLAGS: DWORD =
-        CERT_CHAIN_POLICY_IGNORE_END_REV_UNKNOWN_FLAG |
-        CERT_CHAIN_POLICY_IGNORE_CTL_SIGNER_REV_UNKNOWN_FLAG |
-        CERT_CHAIN_POLICY_IGNORE_CA_REV_UNKNOWN_FLAG |
-        CERT_CHAIN_POLICY_IGNORE_ROOT_REV_UNKNOWN_FLAG;
-pub const CERT_TRUST_HAS_EXACT_MATCH_ISSUER: DWORD = 0x00000001;
-pub const CERT_TRUST_HAS_KEY_MATCH_ISSUER: DWORD = 0x00000002;
-pub const CERT_TRUST_HAS_NAME_MATCH_ISSUER: DWORD = 0x00000004;
-pub const CERT_TRUST_IS_SELF_SIGNED: DWORD = 0x00000008;
-pub const CERT_TRUST_HAS_PREFERRED_ISSUER: DWORD = 0x00000100;
-pub const CERT_TRUST_HAS_ISSUANCE_CHAIN_POLICY: DWORD = 0x00000200;
-pub const CERT_TRUST_HAS_VALID_NAME_CONSTRAINTS: DWORD = 0x00000400;
-pub const CERT_TRUST_IS_COMPLEX_CHAIN: DWORD = 0x00010000;
+pub type PCERT_PRIVATE_KEY_VALIDITY = *mut CERT_PRIVATE_KEY_VALIDITY;
+STRUCT!{struct CERT_KEY_ATTRIBUTES_INFO {
+    KeyId: CRYPT_DATA_BLOB,
+    IntendedKeyUsage: CRYPT_BIT_BLOB,
+    pPrivateKeyUsagePeriod: PCERT_PRIVATE_KEY_VALIDITY,
+}}
+pub type PCERT_KEY_ATTRIBUTES_INFO = *mut CERT_KEY_ATTRIBUTES_INFO;
+pub const CERT_DIGITAL_SIGNATURE_KEY_USAGE: DWORD = 0x80;
+pub const CERT_NON_REPUDIATION_KEY_USAGE: DWORD = 0x40;
+pub const CERT_KEY_ENCIPHERMENT_KEY_USAGE: DWORD = 0x20;
+pub const CERT_DATA_ENCIPHERMENT_KEY_USAGE: DWORD = 0x10;
+pub const CERT_KEY_AGREEMENT_KEY_USAGE: DWORD = 0x08;
+pub const CERT_KEY_CERT_SIGN_KEY_USAGE: DWORD = 0x04;
+pub const CERT_OFFLINE_CRL_SIGN_KEY_USAGE: DWORD = 0x02;
+pub const CERT_CRL_SIGN_KEY_USAGE: DWORD = 0x02;
+pub const CERT_ENCIPHER_ONLY_KEY_USAGE: DWORD = 0x01;
+pub const CERT_DECIPHER_ONLY_KEY_USAGE: DWORD = 0x80;
+STRUCT!{struct CERT_POLICY_ID {
+    cCertPolicyElementId: DWORD,
+    rgpszCertPolicyElementId: *mut LPSTR,
+}}
+pub type PCERT_POLICY_ID = *mut CERT_POLICY_ID;
+STRUCT!{struct CERT_KEY_USAGE_RESTRICTION_INFO {
+    cCertPolicyId: DWORD,
+    rgCertPolicyId: PCERT_POLICY_ID,
+    RestrictedKeyUsage: CRYPT_BIT_BLOB,
+}}
+pub type PCERT_KEY_USAGE_RESTRICTION_INFO = *mut CERT_KEY_USAGE_RESTRICTION_INFO;
+STRUCT!{struct CERT_OTHER_NAME {
+    pszObjId: LPSTR,
+    Value: CRYPT_OBJID_BLOB,
+}}
+pub type PCERT_OTHER_NAME = *mut CERT_OTHER_NAME;
+UNION2!{union CERT_ALT_NAME_ENTRY_u {
+    [u32; 2] [u32; 5],
```

Comment:
According to local testing, this should actually be `[u32; 2] [u64; 2]` or simply `[usize; 2]`.